src/domain/jwt/jwt.repository.ts

```diff
@@ -0,0 +1,30 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { z } from 'zod';
+import { IJwtService } from '@/datasources/jwt/jwt.service.interface';
+import { IJwtRepository } from '@/domain/jwt/jwt.repository.interface';
+
+@Injectable()
+export class JwtRepository implements IJwtRepository {
+  constructor(
+    @Inject(IJwtService)
+    private readonly jwtService: IJwtService,
+  ) {}
+
+  signToken<T extends string | object>(
+    payload: T,
+    options?: {
+      expiresIn?: number;
+      notBefore?: number;
+    },
+  ): string {
+    return this.jwtService.sign(payload, options);
+  }
+
+  verifyToken<T extends z.ZodTypeAny>(
+    accessToken: string,
+    schema: T,
+  ): z.ZodType<z.infer<T>> {
```

Comment:
We are leaking Zod types in a domain interface. Can we use more concrete types here? (as in types that we know that we will be able to handle)

Reply:
I changed this in e0ee482bfd60773bb9423aac0cb5252b157da667 to explicitly use the `JwtAccessTokenSchema` so we can expect the concrete type.